### PR TITLE
add javafx library

### PR DESCRIPTION
**Pull request by klugem**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source <-- not build from source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

For me it would be great to have javafx in conda-forge. 
This recipe downloads packaged jar files from Gluon and copies them into /share/javafx-${VERSION}.
If there is a better way to achieve this or already an existing recipe for javafx, please let me know.

### CODE REVIEW

File: recipes/javafx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  sha256: 40ef06cd50ea535d45403d9c44e9cb405b631c547734b5b50a6cb7b222293f97 # [unix and x86_64]
+  sha256: e98158812db1a0037cdaf85824adff384e41e3edf046fda145479ce6057cb514 # [osx and x86_64]
+  fn: javafx-{{ version }}-sdk.zip 
+
+build:
+  number: 0
+
+requirements:
+  build:
+    skip: true # [win]
```
Comment (marcelotrevisani) [inline recipes/javafx/meta.yaml]: the skip should be in the another build

File: recipes/javafx/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  url: http://gluonhq.com/download/javafx-{{ version }}-sdk-linux
+  fn: javafx-{{ version }}-sdk-linux.zip
+  sha256: 40ef06cd50ea535d45403d9c44e9cb405b631c547734b5b50a6cb7b222293f97
+
+build:
+  number: 0
+
+requirements:
+  build:
+   - coreutils >=8
```
[the remark this answers is not in the document]
Comment (klugem) [reply recipes/javafx/meta.yaml]: Won't hurt. I'll add it.

File: recipes/javafx/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  sha256: e98158812db1a0037cdaf85824adff384e41e3edf046fda145479ce6057cb514  # [osx and x86_64]
+  fn: javafx-{{ version }}-sdk.zip 
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - coreutils >=8
```
Comment (marcelotrevisani) [inline recipes/javafx/meta.yaml]: I think you can remove this ``build`` here, just let the host and run it should be fine

**Comment by isuruf** [comment]
How are you going to use the jar files?

Also see https://github.com/conda-forge/conda-forge.github.io/issues/590

**Comment by klugem** [comment]
> How are you going to use the jar files?

I will use them in a recipes I have in bioconda (watchdog-wms) to allow users to load the graphical interface part of Watchdog.

> Also see [conda-forge/conda-forge.github.io#590](https://github.com/conda-forge/conda-forge.github.io/issues/590)

I saw that before but as far I understood there will no short term solution for that problem. 

However, it might be a good idea to rename the package to java-javafx to be inline with the e [namespaces proposal](https://gist.github.com/mcg1969/da5aec380d2ed083b79ddcf151ca16f19) ?

File: recipes/javafx/meta.yaml (hunk 3 above)
Comment (klugem) [reply recipes/javafx/meta.yaml]: done.

**Comment by isuruf** [comment]
> However, it might be a good idea to rename the package to java-javafx to be inline with the e namespaces proposal ?

Yes.

**Comment by isuruf** [comment]
The binaries were built for GLIBC 2.17, but we require only GLIBC 2.12 (CentOS 6)
You'll have to build from source to get centos 6 compatibility.

**Comment by klugem** [comment]
> The binaries were built for GLIBC 2.17, but we require only GLIBC 2.12 (CentOS 6)
> You'll have to build from source to get centos 6 compatibility.

I think some of the dependencies for building javafx are missing in conda. Would it be ok to use a CentOS 6 docker image and build javafx within the docker image?

**Comment by scopatz** [comment]
> Would it be ok to use a CentOS 6 docker image and build javafx within the docker image?

The short answer is that we require all of the deps to be on conda-forge: https://github.com/conda-forge/staged-recipes/#6-do-all-of-my-packages-dependencies-have-to-be-in-conda-forge-already

**Comment by dbast** [comment]
Downloading java artefacts including dependency resolution and download verification is quite an extended topic. 

Other solution would be to use coursier (tiny script that bootstraps itself) to bridge the gap into the java world (see https://get-coursier.io/docs/cli-overview), the command could be in this case e.g. `./coursier fetch org.openjfx:javafx-base:11.0.2 --cache $CONDA_PREFIX/lib/java/` 

It also allows to run stuff (the BFG git repo cleaner): `./coursier launch com.madgag:bfg:1.12.6 -- -h` ... (does dependency resolution, download, verify, detects main class, runs main class).

Was already thinking to package coursier for conda-forge... Even writing a conda extension to support it in the environment.yaml would be nice:

```
dependencies:
- python
- pytest
- pip:
  - quinn
- coursier:
  - com.madgag:bfg:1.12.6
```

**Comment by klugem** [comment]
> > Would it be ok to use a CentOS 6 docker image and build javafx within the docker image?
> 
> The short answer is that we require all of the deps to be on conda-forge: https://github.com/conda-forge/staged-recipes/#6-do-all-of-my-packages-dependencies-have-to-be-in-conda-forge-already

Thats what I feared. ;)
Building javafx requires (in Ubuntu) these packages (not all part of conda)...:

- libavformat-ffmpeg57
- libgl1-mesa-dev
- libx11-dev 
- pkg-config 
- x11proto-core-dev 
- mercurial 
- git
- libgtk2.0-dev 
- libgtk-3-dev

**Comment by scopatz** [comment]
Most of those actually are part of conda-forge already

**Pull request closed without merging**